db/migrate/20190108115333_add_referral_id_to_users_table.rb

```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+# Add new field due to referral - affiliate relation
+class AddReferralIdToUsersTable < ActiveRecord::Migration[5.2]
+  def change
+    add_column :users, :referral_id, :integer, after: :password_digest
```

Comment:
user_id is bitint not integer,
also you can place it at the end of the table before timestamps

Reply:
would be better with  `add_index :users, :referral_id`